server/game/core/ongoingEffect/OngoingEffect.ts

```diff
@@ -29,7 +34,24 @@ const EnumHelpers = require('../utils/EnumHelpers');
  *                        and the numerical value of the effect, if any.
  */
 class OngoingEffect {
-    constructor(game, source, properties, effectImpl) {
+    public game: Game;
+    public source: Card;
+    // TODO: Can we make GameObject more specific? Can we add generics to the class for AbilityContext?
+    public matchTarget: GameObject | ((target: GameObject, context: AbilityContext) => boolean);
+    public duration?: Duration;
+    public until: WhenType;
+    public condition: (context?: AbilityContext) => boolean;
+    public sourceLocationFilter: LocationFilter;
+    public canChangeZoneOnce: boolean;
+    public canChangeZoneNTimes: number;
```

Comment:
We can outright remove these canChangeZone properties; tbh I thought I already had

Reply:
Yeah go ahead, lmk if it breaks anything. I'm pretty sure they're not used anywhere but would be good to know if I'm wrong

Reply:
Didn't break anything, so we're good. Probably.